### PR TITLE
Make `collection-contrib` and `junit` compile with Dotty

### PR DESCRIPTION
**Pull request by allanrenucci**

### CODE REVIEW

File: test/junit/src/test/scala/strawman/collection/LazyZipOpsTest.scala [hunk 1]
```diff
@@ -64,7 +65,7 @@ class LazyZipOpsTest {
 
   @Test
   def lazyZip2_toIterable(): Unit = {
-    val iter: Iterable[(Int, Int)] = zipped2
+    val iter: collection.Iterable[(Int, Int)] = zipped2
```
Comment (julienrf) [inline test/junit/src/test/scala/strawman/collection/LazyZipOpsTest.scala]: Why was this change needed?

File: test/junit/src/test/scala/scala/collection/mutable/MutableListTest.scala [hunk 2]
```diff
@@ -1,37 +0,0 @@
...
-    val it = is.iterator
-    val x  = Some(3)
-    is += x
-    assertNotReachable(x, it) {
-      it.next()
-      it.next()
-    }
-    assertTrue(it.isEmpty)
-  }
-}
```
Comment (julienrf) [inline test/junit/src/test/scala/scala/collection/mutable/MutableListTest.scala]: Why did you have to remove this file?

Comment (allanrenucci) [reply test/junit/src/test/scala/scala/collection/mutable/MutableListTest.scala]: I removed:
```scala
def assertSameElements[A, B >: A](expected: IterableLike[A, _], actual: GenIterable[B], message: String = ""): Unit
def assertSameElements[A, B >: A](expected: IterableLike[A, _], actual: Iterator[B]): Unit =
```
from `AssertUtil` which were causing ambiguous overloading issues in Dotty. This test relied on them.

File: test/junit/src/test/scala/strawman/collection/LazyZipOpsTest.scala (hunk 1 above)
Comment (allanrenucci) [reply test/junit/src/test/scala/strawman/collection/LazyZipOpsTest.scala]: This is due to a difference in naming resolution. `strawman.collection.immutable._` is imported so here Dotty resolve `Iterable` to `strawman.collection.immutable.Iterable` but `zipped2` returns `strawman.collection.LazyZip2` which is implicitly converted to `View` which is a subtype of `strawman.collection.Iterable` and not `strawman.collection.immutable.Iterable`

File: test/junit/src/test/scala/scala/collection/mutable/MutableListTest.scala (hunk 2 above)
Comment (allanrenucci) [reply test/junit/src/test/scala/scala/collection/mutable/MutableListTest.scala]: Sure I can do that. I removed this file since it was testing something that was not part of the strawman collection anyway.

Comment (julienrf) [reply test/junit/src/test/scala/scala/collection/mutable/MutableListTest.scala]: Yes, you are right but I prefer keeping these files so that we immediately see what has been migrated to the strawman and what hasn’t.
